internal/services/openai.go

```diff
@@ -108,32 +114,44 @@ func (o *openAI) askChatGPT(body io.Reader) (*ChatGPTResponse, error) {
 	return cResp, nil
 }
 
-func (o *openAI) GetErrorCodesDescription(make, model string, errorCodes []string) (string, error) {
+func (o *openAI) GetErrorCodesDescription(make, model string, errorCodes []string) ([]ErrorCodesResponse, error) {
 	codes := strings.Join(errorCodes, ", ")
+
 	req := fmt.Sprintf(`{
 		"model": "gpt-3.5-turbo",
 		"messages": [
 			{
 				"role": "user", 
-				"content": "Briefly summarize for me, in order, what the following error codes mean for a %s %s. The error codes are %s."}]
+				"content": "A %s %s is returning error codes %s. Respond only with a line for each code, in the format 'code:explanation \n'"}]
 	  		}
 	  `, make, model, codes)
 
 	r, err := o.askChatGPT(strings.NewReader(req))
 	if err != nil {
-		return "", err
+		return nil, err
 	}
 
 	appmetrics.OpenAITotalTokensUsedOps.Add(float64(r.Usage.TotalTokens))
 
 	if len(r.Choices) == 0 {
-		return "", errors.New("could not fetch description for error codes")
+		return nil, errors.New("could not fetch description for error codes")
 	}
 
 	c := r.Choices[0]
 	if c.FinishReason != "stop" {
 		o.logger.Error().Interface("rawResponse", r).Msg("Unexpected finish_reason from ChatGPT.")
 	}
 
-	return strings.Trim(c.Message.Content, "\n"), nil
+	resp := []ErrorCodesResponse{}
+
+	codesResp := strings.Split(r.Choices[0].Message.Content, "\n")
+	for _, code := range codesResp {
+		cc := strings.Split(code, ":")
+		resp = append(resp, ErrorCodesResponse{
+			Code:        cc[0],
+			Description: cc[1],
```

Comment:
It seems to me that this always has a space at the start. Just from playing around in the web app.